spring-integration-core/src/main/java/org/springframework/integration/gateway/GatewayMethodMetadata.java

```diff
@@ -60,7 +65,9 @@ public void setPayloadExpression(String payloadExpression) {
 	}
 
 	public void setHeaderExpressions(Map<String, Expression> headerExpressions) {
-		this.headerExpressions = headerExpressions;
+		Assert.notNull(headerExpressions, "'headerExpressions' must not be null");
```

Comment:
Do we really care about it being null or should we simply `clear()` and skip `putAll` if null?